msgpack-value/src/test/java/io/camunda/zeebe/msgpack/UnpackedObjectTest.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.msgpack;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.spy;
+
+import io.camunda.zeebe.msgpack.property.BooleanProperty;
+import io.camunda.zeebe.msgpack.property.IntegerProperty;
+import io.camunda.zeebe.msgpack.property.StringProperty;
+import java.nio.ByteBuffer;
+import org.agrona.MutableDirectBuffer;
+import org.agrona.concurrent.UnsafeBuffer;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+public class UnpackedObjectTest {
+
+  @Nested
+  public class GreyBox {
+
+    @Test
+    public void shouldResetObjectBeforeReadingValue() {
+      // given
+      final var property = new StringProperty("property", "default");
+      final var unpackedObject = new UnpackedObject();
+
+      unpackedObject.declareProperty(property);
+
+      final var buffer = new UnsafeBuffer(ByteBuffer.allocate(100));
+
+      unpackedObject.write(buffer, 0);
+
+      final var spyUnpackedObject = spy(unpackedObject);
+
+      // when
+      spyUnpackedObject.wrap(buffer);
+
+      // then
+      final var orderOfInvocations = Mockito.inOrder(spyUnpackedObject);
+      orderOfInvocations.verify(spyUnpackedObject).reset();
+      orderOfInvocations.verify(spyUnpackedObject).read(Mockito.any());
```

Comment:
This test case is not robust because it depends on implementation details. I don't see much value in it.